pkg/cmd/roachtest/schemachange.go

```diff
@@ -289,30 +289,66 @@ func findIndexProblem(
 	return nil
 }
 
-func registerSchemaChangeTPCC(r *registry) {
+func registerSchemaChangeIndexTPCC1000(r *registry) {
 	warehouses := 1000
 	numNodes := 5
 	r.Add(testSpec{
-		Name:    fmt.Sprintf("schemachange/tpcc/warehouses=%d/nodes=%d", warehouses, numNodes),
+		Name:    fmt.Sprintf("schemachange/index/tpcc-%d", warehouses),
 		Nodes:   nodes(numNodes),
 		Timeout: 4 * time.Hour,
 		Run: func(ctx context.Context, t *test, c *cluster) {
 			runTPCC(ctx, t, c, tpccOptions{
 				Warehouses: warehouses,
 				Extra:      "--wait=false --tolerate-errors",
 				During: func(ctx context.Context) error {
-					conn := c.Conn(ctx, 1)
-					start := timeutil.Now()
-					if _, err := conn.Exec(`
-					CREATE UNIQUE INDEX foo ON tpcc.order (o_entry_d, o_w_id, o_d_id, o_carrier_id, o_id);
-				`); err != nil {
-						t.Fatal(err)
-					}
-					c.l.Printf("CREATE INDEX took %s", timeutil.Since(start))
-					return nil
+					return addIndexes(ctx, t, c, []string{
+						`CREATE UNIQUE INDEX ON tpcc.order (o_entry_d, o_w_id, o_d_id, o_carrier_id, o_id);`,
+						`CREATE INDEX ON tpcc.order (o_carrier_id);`,
+						`CREATE INDEX ON tpcc.customer (c_last, c_first);`,
+					})
 				},
-				Duration: 2 * time.Hour,
+				Duration: 1 * time.Hour,
 			})
 		},
 	})
 }
+
+func registerSchemaChangeIndexTPCC100(r *registry) {
+	warehouses := 100
+	numNodes := 3
+	r.Add(testSpec{
+		Name:    fmt.Sprintf("schemachange/index/tpcc-%d", warehouses),
```

Comment:
How about adding these two tests in a for loop so we share some common code. The tests can have the tpcc/size suffix and can have separate timeouts.

Reply:
Done